### ISSUE TITLE
'inference_input_size' hyperparameter issue with Base Recurrent windows size

### ISSUE THREAD

**Issue by PotosnakW** [Nixtla/neuralforecast#532]
**Describe the bug**
The _create_windows function in _base_recurrent.py generates windows that may have inconsistent input sizes which generates an error when creating the torch.vstack of predictions. I believe this is due to the random sampling procedure in the code screenshot provided below. A screenshot of the error is also provided below.

**To Reproduce**
1. Run the example code below
2. See error

**Expected behavior**
The prediction .csv is generated without errors.

**Screenshots**
<img width="1194" alt="Screen Shot 2023-04-19 at 3 34 43 PM" src="https://user-images.githubusercontent.com/71658863/233181154-c5910c85-de17-49f7-a09f-1e81ac95eafa.png">
<img width="687" alt="Screen Shot 2023-04-19 at 3 39 39 PM" src="https://user-images.githubusercontent.com/71658863/233182193-63be298e-5733-4e32-97f2-b19e3a24c0ac.png">

**Desktop (please complete the following information):**
- OS: Springdale Open Enterprise Linux (--> based on Red Hat Enterprise Linux)
- Browser: N/A
- Version: 8.7 Moderna
- 
**Additional context**
**Example Code Below**

import pandas as pd
import numpy as np

from datasetsforecast.long_horizon import LongHorizon

from neuralforecast.auto import AutoRNN, AutoLSTM, AutoTCN
from neuralforecast.core import NeuralForecast
from neuralforecast.losses.pytorch import MAE

import logging
logging.getLogger("pytorch_lightning").setLevel(logging.WARNING)

Y_df, _, _ = LongHorizon.load(directory='./', group='ETTm2')
Y_df['ds'] = pd.to_datetime(Y_df['ds'])

n_time = len(Y_df.ds.unique())
val_size = int(.1 * n_time)
test_size = int(.2 * n_time)
horizon = 96
freq='15min'
    
rnn_config = {'input_size': 2*horizon,
             'inference_input_size': val_size,
            'max_steps':5,
            'val_check_steps': 2,
            'random_seed': 1, 
            'encoder_n_layers':2,
            'encoder_hidden_size':256,
            'decoder_hidden_size': 256,
            'decoder_layers': 2,
            'batch_size': 2,
            }

fcst = NeuralForecast(freq=freq,
                        models=[AutoRNN(h=horizon,
                                    config=rnn_config,
                                    loss=MAE()),
                                    ],)

fcst_df = fcst.cross_validation(df=Y_df, 
                                val_size=val_size,
                                test_size=test_size, 
                                n_windows=None)
